src/cript/api/api.py

```diff
@@ -0,0 +1,164 @@
+import json
+import warnings
+from typing import List, Literal
+
+import requests
+
+from cript.api._valid_search_modes import _VALID_SEARCH_MODES
+from cript.nodes.primary_nodes.primary_base_node import PrimaryBaseNode
+from cript.nodes.primary_nodes.project import Project
+from cript.nodes.supporting_nodes.group import Group
+from cript.nodes.supporting_nodes.user import User
+from cript.api.exceptions import CRIPTConnectionError
+
+
+class API:
+    host: str = ""
+    _token: str = ""
+
+    def __init__(self, host: str, token: str) -> None:
+        """
+        Initialize object with host and token.
+
+        Parameters
+        ----------
+        host : str
+            CRIPT host to connect to such as "https://criptapp.org"
+            if host ends with a "/" such as "https://criptapp.org/"
+            then it strips it to always be uniform
+        token : str
+            CRIPT API Token used to connect to CRIPT
+
+        Warns
+        -----
+        UserWarning
+            If `host` is using "http".
+
+        Raises
+        ------
+        ConnectionError
+            If it cannot connect to CRIPT with the provided host and token
+
+        Returns
+        -------
+        None
+        """
+
+        # strip ending slash to make host always uniform
+        host = host.rstrip("/")
+
+        self.host = host
+        self._token = token
+
+        # if host is using unsafe "http://" then give a warning
+        if self.host.startswith("http://"):
+            warnings.warn("HTTP is an unsafe protocol please consider using HTTPS")
+
+        # check that api can connect to CRIPT with host and token
+        try:
+            # TODO send an http request to check connection with host and token
+            pass
+        except Exception:
+            raise CRIPTConnectionError
+
+    def save(self, node: PrimaryBaseNode) -> None:
```

Comment:
Even though it is fairly self explanatory, I would add a docstring of the exact specification for each function

Reply:
that is completely fair, and I fully agree. I added a brief docstrings for this but it will likely need more explanation for a graph that is both has nodes to both save and update, but we'll add that when we get to it